usr/share/rear/build/GNU/Linux/400_copy_modules.sh

```diff
@@ -1,28 +1,129 @@
 # 400_copy_modules.sh
 #
-# copy kernel modules for Relax-and-Recover
+# Copy kernel modules to the rescue/recovery system.
+
 #
 # This file is part of Relax-and-Recover, licensed under the GNU General
 # Public License. Refer to the included COPYING for full text of license.
 
-Log "Collecting modules for kernel version $KERNEL_VERSION"
-MODFILES=(
-$( ResolveModules "$KERNEL_VERSION" "${MODULES[@]}" "${MODULES_LOAD[@]}"  )
-)
-StopIfError "Could not resolve kernel module dependancies"
+# The special value MODULES=( 'none' ) enforces that
+# no kernel modules get included in the rescue/recovery system
+# regardless of what modules are currently loaded.
+# Test the first MODULES array element because other scripts
+# in particular rescue/GNU/Linux/240_kernel_modules.sh
+# already appended other modules to the MODULES array:
+if test "none" = "$MODULES" ; then
+    LogPrint "Omit copying kernel modules (MODULES contains 'none')"
+    return
+fi
+
+# Artificial 'for' clause that is run only once
+# to be able to 'continue' with the code after it
+# when the kernel modules have been copied into the rescue/recovery system
+# (the 'for' loop is run only once so that 'continue' is the same as 'break')
+# which avoids dowdy looking code with deeply nested 'if...else' conditions:
+for dummy in "once" ; do
```

Comment:
IMHO you could achieve the same with this code
```shell
if test "none" = "$MODULES" ; then
  LogPrint "Omit copying kernel modules (MODULES contains 'none')"
  return
elif IsInArray "all" "${MODULES[@]}" ; then
  ... copy all modules ...
elif IsInArray "loaded" "${MODULES[@]}" ; then
  ... copy loaded modules ...
else
  for module in "${MODULES[@]}" ; do
    ... copy individual modules ...
  done
fi
```

and IMHO this reads much easier than the fake `for` loop.